content/docs/learning-guide/tutorials/custom-tools/shape-example-cpp.md

```diff
@@ -0,0 +1,383 @@
+
+2. Send a request to create a new entity by calling `EditorRequestBus::BroadcastResult(...)`. Dispatch the `EditorRequests::CreateNewEntity` event, which instantiates an `AZ::EntityId` and stores it in `newEntityId`. 
+
+3. If the user entered a name in the input field, update the entity's name. 
+
+   - Get the name of the entity by calling `m_nameInput->text()` and store it in `entityName`.
+
+   - If the user enabled the check box to apply a suffix of the component's name to the entity's name, query the component's name. Call `EditorComponentAPIBus::BroadcastResult(...)` and dispatch the `EditorComponentAPIRequests::FindComponentTypeNames` event, which finds the component names of the provided list and stores them in `componentNames`.  
+
+   - Format the name and suffix, if any.
+
+    - Call `EditorEntityAPIBus::Event(...)` and dispatch `EditorEntityAPIRequests::SetName` to set the name of `newEntityId` to `entityName`. 
+
+4. Add a Shape component to the entity. Call `EditorComponentAPIBus::Broadcast(...)` and dispatch `EditorComponentAPIRequests::AddComponentsOfType`, which adds the components from the provided list to `newEntityId`.
+
+```cpp
+    void ShapeExampleWidget::CreateEntityWithShapeComponent(const AZ::TypeId& typeId)
+    {
+        // 1
+        using namespace AzToolsFramework;
+
+        // 2
+        AZ::EntityId newEntityId;
+        EditorRequestBus::BroadcastResult(newEntityId, &EditorRequests::CreateNewEntity, AZ::EntityId());
+
+        // 3
+        if (!m_nameInput->text().isEmpty())
+        {
+            QString entityName = m_nameInput->text();
+
+            if (m_addShapeNameSuffix->isChecked())
+            {
+                AZStd::vector<AZStd::string> componentNames;
+                EditorComponentAPIBus::BroadcastResult(
+                    componentNames,
+                    &EditorComponentAPIRequests::FindComponentTypeNames,
+                    AZ::ComponentTypeList{ typeId }
+                );
+
+                AZ_Assert(componentNames.size() == 1, "Unable to find component name");
+
+                QString shapeName(componentNames[0].c_str());
+                shapeName.replace(" ", "");
+
+                entityName = QString("%1_%2").arg(entityName).arg(shapeName);
+            }
+            
+            EditorEntityAPIBus::Event(newEntityId, &EditorEntityAPIRequests::SetName, entityName.toUtf8().constData());
+        }
+
+        // 4
+        EditorComponentAPIBus::Broadcast(&EditorComponentAPIRequests::AddComponentsOfType, newEntityId, AZ::ComponentTypeList{ typeId });
+    }
+```
+
+
+## Icon
+
+An icon is an image file that's used to represent your tool in the O3DE Editor. You will store the icon using the Qt Resource System and load it from your Gem module. We recommend that your icon adheres to the guidelines in [UI development best practices](https://o3de.org/docs/tools-ui/uidev-component-development-guidelines/#ui-development-best-practices).
+
+### Add an icon
+
```

Comment:
I would love to see screenshots under this headings of what the heading is referring to. E.g. Icon